pxr/usdImaging/usdImagingGL/drawModeAdapter.cpp

```diff
@@ -1095,8 +1095,51 @@ UsdImagingGLDrawModeAdapter::_ComputeExtent(UsdPrim const& prim) const
 
     TfTokenVector purposes = { UsdGeomTokens->default_, UsdGeomTokens->proxy,
                                UsdGeomTokens->render };
-    UsdGeomBBoxCache bboxCache(UsdTimeCode::EarliestTime(), purposes, true);
-    return bboxCache.ComputeUntransformedBound(prim).ComputeAlignedBox();
+
+    if (prim.IsLoaded()) {
+        UsdGeomBBoxCache bboxCache(
+            UsdTimeCode::EarliestTime(), purposes, true);
+        return bboxCache.ComputeUntransformedBound(prim).ComputeAlignedBox();
+    } else {
+        GfRange3d extent;
+        UsdAttribute attr;
+        VtVec3fArray extentsHint;
+        // Get the extent either from the authored extent attribute of a
+        // UsdGeomBoundable prim, or get the extentsHint attribute from the
+        // prim.
+        if (prim.IsA<UsdGeomBoundable>() &&
+            (attr = UsdGeomBoundable(prim).GetExtentAttr()) &&
+            attr.Get(&extentsHint, UsdTimeCode::EarliestTime()) &&
+            extentsHint.size() == 2) {
+            extent = GfRange3d(extentsHint[0], extentsHint[1]);
+        }
+        else if ((attr = UsdGeomModelAPI(prim).GetExtentsHintAttr()) &&
```

Comment:
Oof... Can you leave an XXX comment that ideally we'd push the "compute unloaded" bit into UsdGeomBBoxCache, and therby not need to replicate all this logic?

Reply:
Yeah, I was pretty uncomfortable using the std::find over purposes in here, which is part of why I didn't want to push this into UsdGeomBBoxCache myself. But it was the only way I could think of to achieve generality in the face of the possibility of GetOrderedPurposeTokens changing in some way.

Come to think of it, the test for size() < 8 is an unnecessary hard coding of a value, so I'll change that too... Would you suggest completely removing the test for an upper bound on the size of extentsHint, or test that `size() <= purposeTokens.size() * 2`?

Reply:
Yeah, given the way we've defined extentsHint, and the iteration you're doing through purposeTokens, I don't think you need an upper bounds check at all.